### DISCUSSION TITLE
ConditionallyLoadsAttributes works not how it's expected

### DISCUSSION THREAD

**Discussion by Troum** [laravel/framework#52307] [Q&A]
### Laravel Version

10.10

### PHP Version

8.2

### Database Driver & Version

MariaDB

### Description

Hi

During the working process with **Laravel Framework**, especially with **Laravel Eloquent API Resources**, I found that **ConditionallyLoadsAttributes's** methods `when()`,`mergeWhen()`, `whenLoaded()` and others `when*()` methods work not as they are expected to: even if condition is false callback function will fire. 

Please, make logic works as expected

### Steps To Reproduce

1. Create any **Eloquent API Model**
2. Create **Eloquent API Resource** for previously created model
3. Include any condition inside previously created resource using `when()` or `when<whatever>(`) methods
4. Do something inside callback for this `when()`
5. Callback fires even if condition is false

**Comment by iazaran** [comment]
More information and sample codes are needed, but:

If you are using your function like
```php
'secret' =>  $this->when($isAdmin, 'secret-value'),
```
Please change it to the closure like:
```php
'secret' => $this->when($isAdmin, function () {
    return 'secret-value';
}),
```
Then you will ensure that `$isAdmin` will be evaluated before call.

**Reply by Troum** [reply]
Nope, I meant next code:

```
'secret' => $this->when($isAdmin, function () {
    return 'secret-value';
}), 
```

-- this closure will fire even if $isAdmin parameter is false. Independently of true or false - every time closure fires
